src/sdk/bucketspace_fs.js

```diff
@@ -485,13 +496,49 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
     // BUCKET TAGGING //
     ////////////////////
 
-    async put_bucket_tagging(params) {
+    /**
+     * @param {*} params 
+     * @param {nb.ObjectSDK} object_sdk 
+     */
+    async put_bucket_tagging(params, object_sdk) {
         try {
             const { name, tagging } = params;
             dbg.log0('BucketSpaceFS.put_bucket_tagging: Bucket name, tagging', name, tagging);
-            const bucket = await this.config_fs.get_bucket_by_name(name);
-            bucket.tag = tagging;
-            await this.config_fs.update_bucket_config_file(bucket);
+            const bucket_path = this.config_fs.get_bucket_path_by_name(name);
+
+            const bucket_lock_file = `${bucket_path}.lock`;
+            await native_fs_utils.lock_and_run(this.fs_context, bucket_lock_file, async () => {
```

Comment:
so lock_and_run() will create .lock file per bucket, do we want to use it in all our config updates (accounts/buckets etc)? 
if so, can you open a github issue for that? mentioning this requires changes in the CLI and in bucketspace_fs.

Reply:
I don't think all of the ops require locks. Most of them are indeed overwrites however for some of them, especially the ones which are both CLI and AWS S3 API controllable, I think we must have a lock to make sure no races result in reverting the other operation.

Yes, I will open an issue.